tools/istio-iptables/pkg/dependencies/implementation_linux.go

```diff
@@ -261,22 +261,22 @@ func (r *RealDependencies) executeXTablesWithOutput(log *log.Scope, cmd constant
 			c = exec.Command(cmdBin, args...)
 		}
 	}
-	log.Infof("Running command (%s): %s %s", mode, cmdBin, strings.Join(args, " "))
+	if quietLogging {
+		log.Debugf("Running command (%s): %s %s", mode, cmdBin, strings.Join(args, " "))
+	} else {
+		log.Infof("Running command (%s): %s %s", mode, cmdBin, strings.Join(args, " "))
+	}
 
 	c.Stdout = stdout
 	c.Stderr = stderr
 	c.Stdin = stdin
 	err := run(c)
 	if len(stdout.String()) != 0 {
-		if !silenceOutput {
-			log.Infof("Command output: \n%v", stdout.String())
-		} else {
-			log.Debugf("Command output: \n%v", stdout.String())
-		}
+		log.Debugf("Command output: \n%v", stdout.String())
 	}
 
 	// TODO Check naming and redirection logic
-	if (err != nil || len(stderr.String()) != 0) && !ignoreErrors {
+	if (err != nil || len(stderr.String()) != 0) && !quietLogging {
```

Comment:
if `stderr` != nil can we just append it to an errors list and just return `errors.Join` from this func?

Then (again) caller will get all errors, and can decide what to do with them.

Reply:
Not sure if always returning errors based on stderr is safe.
The executed program could print something (like a warning) to stderr but still succeed in its execution (return code 0).
If we always return stderr using errors.Join(), the caller has no guarantee whether the iptables modification succeeded or not.

I think we would need the function to return the output, return code, and errors for the caller to have all the necessary information for handling

Reply:
Yeah - what I am getting at is that `stderror` is either

- Something we should unconditionally print, if > 0, at `error` level.
or
- Something we should unconditionally print, if `err != nil`, at error level.
or
- Something we shouldn't print at all, and should return to the caller if `err != nil`.

and in none of those cases is it particularly useful to have to toggled by the passed-in flag.

Reply:
I think it's probably fine to `join` the stderr as extra error context if the `Start()` err != nil.

If the `Start()` err == nil, at most it's worth a debug.

Reply:
Ah I misunderstood, I get it now :)

I changed to always return the join of the err of the run()+stderr (if err!=nil), regardless of the flag.
The logs will continue to print (at info/debug) the "transformedStderr", which is basically a merge of the error with stderr, along with some special wording to make it more readable.
stderr is now also printed in debug mode in case err is nil.